servicetalk-tcp-netty-internal/src/main/java/io/servicetalk/tcp/netty/internal/TcpServerBinder.java

```diff
@@ -169,6 +175,84 @@ protected void handleSubscribe(Subscriber<? super ServerContext> subscriber) {
         };
     }
 
+    /**
+     * Wraps the connection function with early and late acceptors.
+     *
+     * @param connection the connection function which establishes the pipeline.
+     * @param channel the netty channel on which the connection is run.
+     * @param executionContext The {@link ExecutionContext} to use for the bind socket.
+     * @param earlyConnectionAcceptor the {@link EarlyConnectionAcceptor} to filter newly accepted sockets early.
+     * @param lateConnectionAcceptor the {@link LateConnectionAcceptor} to filter newly accepted sockets.
+     * @param connectionAcceptor The {@link ConnectionAcceptor} used to filter newly accepted sockets.
+     * @param <CC> The type of {@link ConnectionContext} that is created for each accepted socket.
+     * @return the wrapped connection function with the connection acceptors (if any).
+     */
+    private static <CC extends ConnectionContext> Single<CC> wrapConnectionAcceptors(
+            Single<CC> connection,
+            final Channel channel,
+            final ExecutionContext<?> executionContext,
+            @Nullable final EarlyConnectionAcceptor earlyConnectionAcceptor,
+            @Nullable final LateConnectionAcceptor lateConnectionAcceptor,
+            @Nullable final InfluencerConnectionAcceptor connectionAcceptor) {
+        final Executor offloadExecutor = executionContext.executor();
+
+        if (earlyConnectionAcceptor != null) {
+            ReducedConnectionInfo info = new ReducedConnectionInfo() {
```

Comment:
It feels like only `sslSession()` will always be null for `EarlyConnectionAcceptor`. Other fields can be useful to make the decision:
- `sslConfig` signals if the connection is going to have a handshake or not. We can say that it's always known when you build the server, but in practice users responsible for `EarlyConnectionAcceptor` and the server builder can be different people. Another use-case is that `SslConfig` can be dynamic and return different SSL options over time. It's nice to have access to it from the acceptor.
- `socketOptions` - these can also be dynamic in practice (not something that we offer right now, but we can in the future) and influence decisions;
- `protocol` - server can listen for both UDP/QUIC and TCP when we add HTTP/3 support. Users can decide to accept only a certain amount of connections for each protocol;

WDYT if we reuse the fill `ConnectionInfo` interface here?